dimagi/pages/templates/amp/left.amp.html

```diff
@@ -0,0 +1,42 @@
+{% load i18n %}
+{% load dimagidotcom %}
+
+<div class="hide-until-large display-flex flex-valign-middle" [class]="nav_hide1">
+  <a class="menu-open-sibling {% click_nav 'products' %}"
+     href="{% url 'commcare' %}">
+    {% blocktrans %}
+      Products
+    {% endblocktrans %}
+  </a>
+
+  <a class="menu-open-sibling {% click_nav 'services' %}"
+     href="{% url 'services' %}">
+    {% blocktrans %}
+      Services
+    {% endblocktrans %}
+  </a>
+  <a class="{% click_nav 'more' %}"
+     data-drawer="menu-drawer"
+     role="button"
+     href="#"
+   on="tap:AMP.setState({ drpdwn_cls :'menu-active-amp' , close_btn:'menu-close-amp' ,nav_hide1:'main-nav-content-hide' ,nav_hide2:'main-nav-content-hide'})">
```

Comment:
This is a lot of duplicated code just to add these few AMP-specific lines in. I don't think this is good for maintainability. Can you please find a way through server-side user agent detection to make these smaller changes in the original template?

Reply:
We have not removed menu.js, it is still there just for pillar page instead of menu.js we have used inbuilt amp tags and custom css to make the code more efficient and also performance wise also it's better

Yes there are still some duplicate codes, we have tried to minimize as less as we can. If we are creating separate amp page the duplication of codes will get exceed even more. What we have done is that we have used everything inbuilt of amp and have made it in this way so that no user experience should get disturbed by keeping all the UI and design same.

If we get any server flag to detect whether the the user agent is mobile or desktop that will also helps in duplicating some code.